controllers/helpers.go

```diff
@@ -48,20 +47,10 @@ func pausedPredicates(logger logr.Logger) predicate.Funcs {
 func processIfUnpaused(logger logr.Logger, obj runtime.Object, meta metav1.Object) bool {
 	kind := strings.ToLower(obj.GetObjectKind().GroupVersionKind().Kind)
 	log := logger.WithValues("namespace", meta.GetNamespace(), kind, meta.GetName())
-	if isPaused(nil, meta) {
+	if clusterutil.HasPausedAnnotation(meta) {
```

Comment:
shouldn't this be clusterutil.IsPaused(... ?

Reply:
Ah, nevermind. It's not for the cluster object.

Reply:
`clusterutil.IsPaused` does not handle a nil cluster, so this basically just checks the paused annotation which is what using a nil cluster with the local `isPaused` would fall back to.